combat-trainer.lic

```diff
@@ -171,13 +176,27 @@ class SetupProcess
       game_state.prepare_summoned_weapon(last_summoned)
     else
       @equipment_manager.wield_weapon(game_state.weapon_name, game_state.weapon_skill)
+      if game_state.use_whirlwind? && game_state.is_whirlwind_trainable?(game_state.weapon_skill)
+        determine_whirlwind_weapon(game_state)
+        echo("Combat-Trainer:: Whirlwinding with -=== MAIN: #{game_state.weapon_skill}  OFF: #{game_state.whirlwind_offhand_skill} ===-") if $debug_mode_ct
+        game_state.set_currently_whirlwinding(true)         
+      else
+        game_state.set_currently_whirlwinding(false)
+      end
     end
 
     # Invoke Focus
     return unless game_state.weapon_skill == 'Targeted Magic' && game_state.weapon_name
     bput("invoke #{game_state.weapon_name}", 'You')
     waitrt?
   end
+
+  def determine_whirlwind_weapon(game_state)
+    temp_array = (game_state.whirlwind_trainables - [game_state.weapon_skill])
+    offhand_skill = temp_array.min_by { |skill| [DRSkill.getxp(skill), DRSkill.getrank(skill)] }
+    game_state.update_whirlwind_weapon_info(offhand_skill)
```

Comment:
Does this account for multiple weapon skills using the same weapon (e.g. swappable weapons)?